gr-iio/lib/ad9081_common_impl.cc

```diff
@@ -0,0 +1,106 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ad9081_common_impl.h"
+
+#include <gnuradio/iio/ad9081_common.h>
+
+#include <cmath>
+#include <cstdint>
+
+namespace gr {
+namespace iio {
+
+long long phase_fold(float x)
+{
+    return (long long)(1000 *
+                       (x - static_cast<int64_t>(floor((x + 180.0) / 360.0)) * 360.0));
+}
+
+int64_t nyquist_fold(int64_t f_s, int64_t f_c /*, int64_t& f_lo, bool& odd*/)
+{
+    int64_t f_n = f_s >> 1;
+
+    bool neg = f_c < 0;
+    if (neg)
+        f_c = -f_c;
+
+    int64_t n = (f_c / f_n + 1) >> 1;
+
+    int r = f_c - n * f_s;
```

Comment:
why the sudden (typically narrowing) type change? You're even returning an `int64_t`!